src/modules/banner/banner.service.ts

```diff
@@ -49,6 +50,28 @@ export class BannerService {
         return buildPaginatedResponse(banners, totalItems, pagination)
     }
 
+    async search(
+        companyId: string,
+        pagination: PaginationQuery,
+        filter?: BannerSearchFilterDTO,
+        order?: BannerOrderByDTO,
+        query?: string,
+    ): Promise<BannerResponsePaginatedDTO> {
+        const paginationParams = getPaginatedParams(pagination)
+
+        const [banners, total] =
+            await this.prismaService.banner.findManyAndCount({
+                where: {
+                    companyId: companyId,
+                    ...this.buildSearchParams(query, filter),
```

Comment:
```suggestion
                    ...this.buildSearchParams(query, filter),
                    companyId: companyId,
```
Só inverte a posição aqui, pois o companyId deve sempre manter no where